core/src/main/java/org/geysermc/geyser/registry/populator/ItemRegistryPopulator.java

```diff
@@ -492,7 +599,36 @@ public static void populate() {
 
                 componentBuilder.putCompound("item_properties", itemProperties.build());
                 builder.putCompound("components", componentBuilder.build());
-                furnaceMinecartData = new ComponentItemData("geysermc:furnace_minecart", builder.build());
+                componentItemData.add(new ComponentItemData("geysermc:furnace_minecart", builder.build()));
+
+                // Register any completely custom items given to us
+                Set<Integer> registeredJavaIds = new ObjectOpenHashSet<>(); //Used to check for duplicate item java ids
```

Comment:
Use `IntSet` to avoid boxing.
```suggestion
                IntSet registeredJavaIds = new IntOpenHashSet<>(); // Used to check for duplicate item java ids
```